parser/base_scraper.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+from parser.apartment import Apartment
+
+
+class BaseScraper(ABC):
+
+    def __init__(self):
+        self.apartments_data = []
+
+    @abstractmethod
+    def fetch_apartment_listings(self, page_content) -> list[str]:
```

Comment:
Make it clear that `fetch_apartment_listings` method returns a list of something that is an apartment and is used by all other methods. This will increase code readability. You can do this by introducing some dummy types or to import those types from bf4 module (i don't know if there are some types for import).
```python
from typing import Any

ApartmentPageData = Any  # Any because we don't actually know what is the type of it because a library does not provide an info about it.

...
def fetch_apartment_listings(self, page_content) -> list[ApartmentPageData]: ...

    def get_apartment_title(self, apartment: ApartmentPageData) -> str: ...

```